### PR TITLE
ci: add workflow for compliance tests

### PR DESCRIPTION
**Pull request by utzig**
Add compliance test on GH directly from the Zephyr tree.

### CODE REVIEW

File: .github/workflows/compliance.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Upload Results
+        uses: actions/upload-artifact@master
+        continue-on-error: True
+        with:
+          name: compliance.xml
+          path: ncs/nrfxlib/compliance.xml
+
+      - name: Check Warnings
+        run: |
+          for file in Codeowners.txt Devicetree.txt Gitlint.txt Identity.txt Nits.txt pylint.txt checkpatch.txt Kconfig.txt; do
```
Comment (carlescufi) [inline .github/workflows/compliance.yml:49]: Very nice! Could you update the one in sdk-nrf with this trick too?

File: .github/workflows/compliance.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        working-directory: ncs
+        run: |
+          pip3 install -U pip
+          pip3 install -U setuptools
+          export PATH="$HOME/.local/bin:$PATH"
+          pip3 install -U west
+          west init -m https://github.com/nrfconnect/sdk-nrf
+          west update zephyr
+          export ZEPHYR_BASE=$(pwd)/zephyr
+          pip install -U gitlint
```
Comment (carlescufi) [inline .github/workflows/compliance.yml:35]: pip3?

File: .github/workflows/compliance.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          pip install -r zephyr/scripts/requirements-compliance.txt
```
Comment (carlescufi) [inline .github/workflows/compliance.yml:36]: pip3

File: .github/workflows/compliance.yml (hunk 1 above)
Comment (utzig) [reply .github/workflows/compliance.yml:49]: Sure, will do.

File: .github/workflows/compliance.yml (hunk 2 above)
Comment (utzig) [reply .github/workflows/compliance.yml]: Done.

File: .github/workflows/compliance.yml (hunk 3 above)
Comment (utzig) [reply .github/workflows/compliance.yml]: Done.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**